src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -0,0 +1,222 @@
+from collections import defaultdict
+from functools import reduce
+from operator import or_
+from typing import List, Mapping, MutableMapping, Optional, Sequence, Set, Union
+
+from django.db.models import Q
+
+from sentry.sentry_metrics.indexer.models import StringIndexer as StringIndexerTable
+from sentry.sentry_metrics.indexer.models import indexer_cache
+from sentry.utils import metrics
+from sentry.utils.services import Service
+
+_INDEXER_CACHE_METRIC = "sentry_metrics.indexer.memcache"
+
+
+class PGStringIndexerV2(Service):
+    """
+    Provides integer IDs for metric names, tag keys and tag values
+    and the corresponding reverse lookup.
+    """
+
+    __all__ = ("record", "resolve", "reverse_resolve", "bulk_record")
+
+    def _record_many_strings(self, keys: Sequence[str]) -> Mapping[str, int]:
```

Comment:
Why do you need a dedicated method for this? This method is used only in one place.
Generally if you have a private method that is used in one place only, the code is not easier to follow than if the method was inline where it is used as the additional method does not provide any additional abstraction but some additional complexity when you read the code.

If you try to break down the code of `bulk_record`, a better way to proceed is to identify an abstraction you can reuse and that has a specific responsibility and a specific interface. Put it in a class and use it. 
That helps in keeping your code in a more readable form.